### PR TITLE
Pytest

### PR DESCRIPTION
**Pull request by natalyshevchuk**

### CODE REVIEW

File: pages/companies_page.py [hunk 1]
```diff
@@ -9,45 +11,55 @@ def __init__(self, browser):
...
     def confirm_with_popup(self):
+        """Returns the text from the pop-up message"""
         popup = self.browser.driver.get_element(self.locators.POPUP)
         confirm_popup = popup.find_element_by_tag_name('p').text
         return confirm_popup
 
     def companies_are_visible(self):
+        """Checks the existence of companies"""
         companies = self.browser.get_element_with_time_delay(self.locators.TABLE_BODY)
         if companies != 0:
```
Comment (TanyaKalistruk) [inline pages/companies_page.py:52]: `return companies != 0` ?

File: pages/hot_vacancies_page.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                                           guest_data.HOT_PREVIOUS)
+
+    def details_text(self):
+        """Returns the text of the vacancy"""
+        text_details = self.browser.pop_up_element(self.locators.HOT_VACANCY_INFO).text
+        return text_details
+
+    def next_click_test(self):
+        """Returns text from the vacancy on the next page"""
+        click_next = self.browser.pop_up_element(self.locators.HOT_NEXT_TEST).text
```
Comment (TanyaKalistruk) [inline pages/hot_vacancies_page.py]: where is "click"?

Comment (natalyshevchuk) [reply pages/hot_vacancies_page.py]: Changed

File: pages/user_profile_page.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    def clear_data_textbox(self, key):
+        """Clears specified textbox"""
+        self.browser.clear_element(self.locators.user_fields[key])
+
+    def enter_data_textbox(self, key, value):
+        """Enters into the specific field data and returns True/False for valid/invalid data"""
+        self.clear_data_textbox(key)
+        self.browser.send_keys(self.locators.user_fields[key], value)
+        return True if self.check_validity_data(key) else False
```
Comment (TanyaKalistruk) [inline pages/user_profile_page.py:19]: return self.check_validity_data(key)

File: pages/user_profile_page.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    def check_invalidity_data(self, key):
+        """Verifies if UI alerts about invalid data input"""
+        element_string = self.browser.get_attr_value(self.locators.user_fields[key], 'class')
+        return True if 'ng-invalid' in element_string else False
```
Comment (TanyaKalistruk) [inline pages/user_profile_page.py:24]: return 'ng-invalid' in element_string

File: pages/user_profile_page.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    def check_validity_data(self, key):
+        """Verifies if UI accepts valid data input"""
+        element_string = self.browser.get_attr_value(self.locators.user_fields[key], 'class')
+        return True if 'ng-valid' in element_string else False
```
Comment (TanyaKalistruk) [inline pages/user_profile_page.py:29]: the same

File: pages/user_profile_page.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    def read_data_textbox(self, key):
+        """Extracts text data from specified textbox"""
+        return self.browser.get_attr_value(self.locators.user_fields[key], 'ng-reflect-model')
+
+    def disabled_update_profile_button(self):
+        """Returns True/False if the Update Profile button is disabled/enabled"""
+        button_property = self.browser.get_property_wrapper(self.locators.UPDATE_PROFILE,
+                                                            'disabled')
+        return True if button_property else False
```
Comment (TanyaKalistruk) [inline pages/user_profile_page.py:39]: the same

File: tasks.py [hunk 7]
```diff
@@ -1,5 +1,12 @@
-from invoke import task, run
+"""This module contains tasks for the project"""
+from invoke import task
+
+
+@task
+def pylint(c):
```
Comment (TanyaKalistruk) [inline tasks.py:6]: did you run pylint?

File: tests/test_delete_company.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    with allure.step("Login"):
+        app.header.select_option(CownerData.OPTION)
+        app.sign_in_page.login(CownerData.PERSON)
+    with allure.step("Click the 'All companies' on the header"):
+        app.header.click_my_companies()
+    with allure.step("Delete the company "):
+        app.my_companies_page.click_company_delete(CownerData.COMPANY_DELETE)
+    with allure.step("Check the result"):
+
+        a =app.my_companies_page.check_company_absence(CownerData.COMPANY_DELETE)
```
Comment (TanyaKalistruk) [inline tests/test_delete_company.py]: what is `a`?

File: tests/test_forgot_password.py [hunk 9]
```diff
@@ -1,42 +1,36 @@
...
+""" This module automate recover password for registered user """
 import allure
-# from init import resource_setup
-from config import EMAIL_SUBJECT_PASSW_RECOVERY, USERNAME_PASSW_RECOVERY, \
-    EMAIL_FORGOT_PASSWORD, FROM_FORGOT_PASSWORD, OLD_PASSWORD, NEW_PASSWORD
+from config import USERNAME_PASSW_RECOVERY, OLD_PASSWORD, NEW_PASSWORD
 from utilities.func import login
 from utilities.db import change_varification_link
 
+MOCK_CONFIRMATION_LINK = \
```
Comment (TanyaKalistruk) [inline tests/test_forgot_password.py:7]: is this link always the same?

File: tests/test_sign_up.py [hunk 10]
```diff
@@ -1,103 +1,109 @@
...
+    app.sign_in_page.click_sign_up_tab()
 
-    page.sign_in_page.enter_sign_up_email("user@gmail.com")
-    page.sign_in_page.enter_sign_up_password(PASSWORD)
-    page.sign_in_page.enter_sign_up_matching_password(PASSWORD)
-    page.sign_in_page.click_sing_up_button()
+    app.sign_in_page.enter_sign_up_email(USER_MAIL)
+    app.sign_in_page.enter_sign_up_password(PASSWORD)
+    app.sign_in_page.enter_sign_up_matching_password(PASSWORD)
+    app.sign_in_page.click_sing_up_button()
```
Comment (TanyaKalistruk) [inline tests/test_sign_up.py:49]: some allure steps before the next part?

File: tests/test_sign_up.py [hunk 11]
```diff
@@ -1,103 +1,109 @@
...
+    app.sign_in_page.click_sign_up_tab()
 
-    page.sign_in_page.enter_sign_up_email("user@gmail.com")
-    page.sign_in_page.enter_sign_up_password("yneoi")
-    page.sign_in_page.enter_sign_up_matching_password("yneoi")
-    page.sign_in_page.click_sing_up_button()
+    app.sign_in_page.enter_sign_up_email(USER_MAIL)
+    app.sign_in_page.enter_sign_up_password(PASSWORD_INCORRECT)
+    app.sign_in_page.enter_sign_up_matching_password(PASSWORD_INCORRECT)
+    app.sign_in_page.click_sing_up_button()
```
Comment (TanyaKalistruk) [inline tests/test_sign_up.py:68]: the same with allure steps

File: utilities/func.py [hunk 12]
```diff
@@ -1,11 +1,8 @@
-from pages.auth_page import AuthPage
-# import allure
+"""This module contains login method"""
```
Comment (TanyaKalistruk) [inline utilities/func.py:1]: maybe rename the name of this module to 'login' or something like that?

File: tests/test_delete_company.py (hunk 8 above)
Comment (natalyshevchuk) [reply tests/test_delete_company.py]: Deleted)

File: tasks.py (hunk 7 above)
Comment (natalyshevchuk) [reply tasks.py:6]: yes, everywhere this time:)

**Pull request closed without merging**